lib/resource/resource.ex

```diff
@@ -2334,6 +2336,107 @@ defmodule AshGraphql.Resource do
     String.to_atom(to_string(type) <> "_sort_field")
   end
 
+  def map_definitions(resource, schema, env) do
+    resource = Ash.Type.NewType.subtype_of(resource)
+
+    if AshGraphql.Resource.Info.type(resource) do
+      resource
+      |> get_auto_maps()
+      |> Enum.flat_map(fn attribute ->
+        constraints = Ash.Type.NewType.constraints(attribute.type, attribute.constraints)
+
+        type_name =
+          if constraints[:fields] do
+            if Ash.Type.NewType.new_type?(attribute.type) do
+              cond do
+                function_exported?(attribute.type, :graphql_type, 0) ->
+                  attribute.type.graphql_type()
+
+                function_exported?(attribute.type, :graphql_type, 1) ->
+                  attribute.type.graphql_type(attribute.constraints)
+
+                true ->
+                  map_type(resource, attribute.name)
+              end
+            else
+              map_type(resource, attribute.name)
+            end
+          else
+            nil
+          end
+
+        input_type_name =
+          if constraints[:fields] do
+            if Ash.Type.NewType.new_type?(attribute.type) do
+              cond do
+                function_exported?(attribute.type, :graphql_input_type, 0) ->
+                  attribute.type.graphql_input_type()
+
+                function_exported?(attribute.type, :graphql_input_type, 1) ->
+                  attribute.type.graphql_input_type(attribute.constraints)
+
+                true ->
+                  map_type(resource, attribute.name, _input? = true)
+              end
+            else
+              map_type(resource, attribute.name, _input? = true)
+            end
+          else
+            nil
+          end
+
+        [
+          type_name
+        ]
+        |> Enum.filter(& &1)
+        |> Enum.map(fn type_name ->
```

Comment:
We need to do this differently for input and non-input, since some types have different input representations.